2018-03-05 16:05:55,363 ERROR: Exception on /index [GET] [in /usr/local/lib/python3.6/site-packages/flask/app.py:1560]
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.6/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/local/lib/python3.6/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/Users/evgeny.zemtsov/Desktop/Programming-Languages/Python/Web/app/routes.py", line 26, in index
    return render_template('index.djhtml', title='Home Page', posts=posts)
  File "/usr/local/lib/python3.6/site-packages/flask/templating.py", line 134, in render_template
    context, ctx.app)
  File "/usr/local/lib/python3.6/site-packages/flask/templating.py", line 116, in _render
    rv = template.render(context)
  File "/usr/local/lib/python3.6/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.6/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/Users/evgeny.zemtsov/Desktop/Programming-Languages/Python/Web/app/templates/index.djhtml", line 1, in top-level template code
    {% extends "base.djhtml" %}
  File "/Users/evgeny.zemtsov/Desktop/Programming-Languages/Python/Web/app/templates/base.djhtml", line 31, in top-level template code
    {% block content %}{% endblock %}
  File "/Users/evgeny.zemtsov/Desktop/Programming-Languages/Python/Web/app/templates/index.djhtml", line 6, in block "content"
    {{ form.hidden_tag() }}
  File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 430, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'form' is undefined
2018-03-05 16:06:42,470 INFO: Microblog startup [in /Users/evgeny.zemtsov/Desktop/Programming-Languages/Python/Web/app/__init__.py:48]
2018-03-05 16:06:51,711 INFO: Microblog startup [in /Users/evgeny.zemtsov/Desktop/Programming-Languages/Python/Web/app/__init__.py:48]
2018-03-05 16:06:54,202 INFO: Microblog startup [in /Users/evgeny.zemtsov/Desktop/Programming-Languages/Python/Web/app/__init__.py:48]
2018-03-05 16:06:56,638 ERROR: Exception on /index [GET] [in /usr/local/lib/python3.6/site-packages/flask/app.py:1560]
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.6/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/local/lib/python3.6/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/Users/evgeny.zemtsov/Desktop/Programming-Languages/Python/Web/app/routes.py", line 26, in index
    return render_template('index.djhtml', title='Home Page', posts=posts)
  File "/usr/local/lib/python3.6/site-packages/flask/templating.py", line 134, in render_template
    context, ctx.app)
  File "/usr/local/lib/python3.6/site-packages/flask/templating.py", line 116, in _render
    rv = template.render(context)
  File "/usr/local/lib/python3.6/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.6/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/Users/evgeny.zemtsov/Desktop/Programming-Languages/Python/Web/app/templates/index.djhtml", line 1, in top-level template code
    {% extends "base.djhtml" %}
  File "/Users/evgeny.zemtsov/Desktop/Programming-Languages/Python/Web/app/templates/base.djhtml", line 31, in top-level template code
    {% block content %}{% endblock %}
  File "/Users/evgeny.zemtsov/Desktop/Programming-Languages/Python/Web/app/templates/index.djhtml", line 6, in block "content"
    {{ form.hidden_tag() }}
  File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 430, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'form' is undefined
2018-03-05 16:17:02,678 ERROR: Exception on /index [GET] [in /usr/local/lib/python3.6/site-packages/flask/app.py:1560]
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.6/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/local/lib/python3.6/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/Users/evgeny.zemtsov/Desktop/Programming-Languages/Python/Web/app/routes.py", line 26, in index
    return render_template('index.djhtml', title='Home Page', posts=posts)
  File "/usr/local/lib/python3.6/site-packages/flask/templating.py", line 134, in render_template
    context, ctx.app)
  File "/usr/local/lib/python3.6/site-packages/flask/templating.py", line 116, in _render
    rv = template.render(context)
  File "/usr/local/lib/python3.6/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.6/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/Users/evgeny.zemtsov/Desktop/Programming-Languages/Python/Web/app/templates/index.djhtml", line 1, in top-level template code
    {% extends "base.djhtml" %}
  File "/Users/evgeny.zemtsov/Desktop/Programming-Languages/Python/Web/app/templates/base.djhtml", line 31, in top-level template code
    {% block content %}{% endblock %}
  File "/Users/evgeny.zemtsov/Desktop/Programming-Languages/Python/Web/app/templates/index.djhtml", line 6, in block "content"
    {{ form.hidden_tag() }}
  File "/usr/local/lib/python3.6/site-packages/jinja2/environment.py", line 430, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'form' is undefined
2018-03-05 16:19:33,512 INFO: Microblog startup [in /Users/evgeny.zemtsov/Desktop/Programming-Languages/Python/Web/app/__init__.py:48]
2018-03-05 16:30:01,337 INFO: Microblog startup [in /Users/evgeny.zemtsov/Desktop/Programming-Languages/Python/Web/app/__init__.py:48]
2018-03-05 16:30:35,835 INFO: Microblog startup [in /Users/evgeny.zemtsov/Desktop/Programming-Languages/Python/Web/app/__init__.py:48]
2018-03-05 16:34:43,492 INFO: Microblog startup [in /Users/evgeny.zemtsov/Desktop/Programming-Languages/Python/Web/app/__init__.py:48]
2018-03-05 16:38:15,492 INFO: Microblog startup [in /Users/evgeny.zemtsov/Desktop/Programming-Languages/Python/Web/app/__init__.py:48]
